appinventor/components/src/com/google/appinventor/components/runtime/ChartDataBase.java

```diff
@@ -66,6 +88,21 @@ public void Color(int argb) {
     @SimpleProperty(
             category = PropertyCategory.APPEARANCE)
     public String Label() {
+        try {
+            label = (String) threadRunner.submit(new Callable<Object>() {
```

Comment:
Using the toy example, fix the issue for getters

For some reason, this gets invoked before the Source() setter..